src/ncp/ncp_base.cpp

```diff
@@ -3626,6 +3669,66 @@ ThreadError NcpBase::SetPropertyHandler_POWER_STATE(uint8_t header, spinel_prop_
     return SendLastStatus(header, SPINEL_STATUS_UNIMPLEMENTED);
 }
 
+ThreadError NcpBase::SetPropertyHandler_HOST_POWER_STATE(uint8_t header, spinel_prop_key_t key, const uint8_t *value_ptr,
+                                                         uint16_t value_len)
+{
+    uint8_t value;
+    spinel_ssize_t parsedLength;
+    ThreadError errorCode = kThreadError_None;
+
+    parsedLength = spinel_datatype_unpack(
+                       value_ptr,
+                       value_len,
+                       SPINEL_DATATYPE_UINT8_S,
+                       &value
+                   );
+
+    if (parsedLength > 0)
+    {
+        switch (value)
+        {        
+        case SPINEL_HOST_POWER_STATE_OFFLINE:
+        case SPINEL_HOST_POWER_STATE_DEEP_SLEEP:
+        case SPINEL_HOST_POWER_STATE_LOW_POWER:
+        case SPINEL_HOST_POWER_STATE_ONLINE:
+            // Adopt the requested power state.
+            mHostPowerState = static_cast<spinel_host_power_state_t>(value);
+            break;
+
+        case SPINEL_HOST_POWER_STATE_RESERVED:
+            // Per the specification, treat this as synonymous with SPINEL_HOST_POWER_STATE_DEEP_SLEEP.
+            mHostPowerState = SPINEL_HOST_POWER_STATE_DEEP_SLEEP;
+            break;
+
+        default:
+            // Per the specification, treat unrecognized values as synonymous with SPINEL_HOST_POWER_STATE_LOW_POWER.
+            mHostPowerState = SPINEL_HOST_POWER_STATE_LOW_POWER;
+            break;
+        }
+    }
+    else
+    {
+        errorCode = kThreadError_Parse;
+    }
+
+    if (errorCode == kThreadError_None)
+    {
+        errorCode = HandleCommandPropertyGet(header, key);
```

Comment:
We probably need to handle the case if this returns `NoBuf` error (like `end of scan`)